src/transformState.js

```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let action of actions) {
+    switch (action.type) {
+      case ('clear'):
+        for (const prop in state) {
+          delete state[prop];
+        }
+
+        break;
+
+      case ('addProperties'):
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case ('removeProperties'):
+        for (const prop of action.keysToRemove) {
+          delete state[prop];
+        }
+
+        break;
+
+      default:
+        action = null;
```

Comment:
```suggestion
       throw new Error('Unexpected state');
```
it's better to throw error like this